libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt

```diff
@@ -14,42 +14,69 @@ import retrofit2.Callback
 import retrofit2.Response
 
 @MapboxNavigationModule(MapboxNavigationModuleType.OffboardRouter, skipConfiguration = true)
-class MapboxOffboardRouter(
-    private val context: Context,
-    private val mapboxToken: String
-) : Router {
+class MapboxOffboardRouter : Router {
+
+    companion object {
+        const val ERROR_FETCHING_ROUTE = "Error fetching route"
+    }
 
+    private val context: Context
+    private val mapboxToken: String
+    private val routeBuilderProvider: () -> NavigationRoute.Builder
     private var navigationRoute: NavigationRoute? = null
 
+    constructor(
+        context: Context,
+        mapboxToken: String
+    ) : this(
+        context,
+        mapboxToken,
+        {
+            NavigationRoute
+                .builder(context)
+                .accessToken(mapboxToken)
+        }
+    )
+
+    internal constructor(
+        context: Context,
+        mapboxToken: String,
+        routeBuilderProvider: () -> NavigationRoute.Builder
+    ) {
+        this.context = context
+        this.mapboxToken = mapboxToken
+        this.routeBuilderProvider = routeBuilderProvider
+    }
+
     override fun getRoute(
         origin: Point,
         waypoints: List<Point>,
         destination: Point,
         callback: Router.Callback
     ) {
-        val builder = NavigationRoute
-            .builder(context)
-            .accessToken(mapboxToken)
+        val builder = routeBuilderProvider.invoke()
+        builder
```

Comment:
NIT: This line can be skipped.

Reply:
Hmmm... It is very interesting, but when I removed this line -> `builder` is created and even after setting `origin` and `destination` to it -> it still empty builder) Honestly, I don't know why `routeBuilderProvider.invoke().origin(origin).destination(destination)` creates empty builder... 
So I revert it back

Reply:
@LukasPaczos

Reply:
Looks like each invocation of a builder method is returning a different instance as `this`. Weird.

Reply:
:eyes: https://github.com/mapbox/mapbox-navigation-android/pull/2240#discussion_r351040760

Reply:
Those are ultimately equal statements @Guardiola31337. I'd be more interested in why directly using an instance created by the lambda results in different instances being returned.